pkg/engine/engine.go

```diff
@@ -47,7 +47,25 @@ func (e Engine) Render(templates map[string][]byte, values map[string]interface{
 
 		// TODO: Could we detect unused variables, and give warning about those?
 
-		rendered[name] = []byte(output)
+		// File names can be templates to, render them here as well
+
+		buf.Reset()
+
+		template, err := t.New("__template_file_filename").Parse(name)
```

Comment:
Seems that it is not an issue to use same name on every round of the loop. Name is mandatory even though it is not really needed when the template instance is at hand here